server/models/Admin.js

```diff
@@ -0,0 +1,26 @@
+const mongoose = require("mongoose")
+const bcrypt = require("bcryptjs")
+
+const AdminSchema = mongoose.Schema({
+  username: { type: String, required: true, index: { unique: true } },
+  password: { type: String, required: true },
+})
+
+AdminSchema.pre("save", function(next) {
+  bcrypt
+    .hash(this.password, 8)
+    .then(hashedPassword => {
+      this.password = hashedPassword
+
+      return next()
+    })
+    .catch(err => {
+      next(err)
+    })
+})
+
+AdminSchema.methods.validPassword = function(password) {
+  return bcrypt.compareSync(password, this.password)
```

Comment:
This is also throwing up a linter warning. It is saying that we should use the async method. What do you think?

Reply:
it doesn't a lot sense but since we're going to use async/await it will look nicer